drafts/str4d-stratum/draft1.rst

```diff
@@ -0,0 +1,256 @@
+Abstract
+========
+
+This ZIP describes the Zcash variant of the Stratum protocol, used by miners to
+communicate with mining pool servers.
+
+
+Specification
+=============
+
+``mining.subscribe()``
+----------------------
+
+Request::
+
+    {"id": 1, "method": "mining.subscribe", "params": ["CONNECT_HOST", CONNECT_PORT, "MINER_USER_AGENT", "SESSION_ID"]}\n
+
+- ``CONNECT_HOST`` (str): The host string used by the miner.
+
+  - ``pool.example.com``
```

Comment:
It has the same use cases as in HTTP protocol. Server also cannot "trust" it, but it can give useful information about virtual host (pool). Consider virtual pools and private URLs (DDoS protection), which ends up aggregated on stratum backends.